extpreimage/client.go

```diff
@@ -0,0 +1,139 @@
+package extpreimage
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"google.golang.org/grpc"
+	grpcpool "github.com/processout/grpc-go-pool"
+)
+
+// RPC is an interface implemented by the grpc package
+type RPC interface {
+	Dial(host string, opt grpc.DialOption) (*grpc.ClientConn, error)
+	WithInsecure() grpc.DialOption
+	NewClient(*grpcpool.ClientConn) (ExternalPreimageServiceClient)
+}
+
+// grpcRpc exposes the methods from the grpc package that we need
+// this allows us to stub out the grpc methods more easily
+type grpcRpc struct {}
+
+func (r *grpcRpc) Dial(host string, opt grpc.DialOption) (*grpc.ClientConn, error) {
+  return grpc.Dial(host, opt)
+}
+
+func (r *grpcRpc) WithInsecure() grpc.DialOption {
+  return grpc.WithInsecure()
```

Comment:
[off-topic] do we have a ticket for auth between lnd and the interchain router?

Reply:
yes, we have one: https://trello.com/c/yzqszsum/385-secure-connection-between-lightning-client-and-broker-interchain-router